frontend/src/cards/ui/TwoLevelDropDown.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react";
+import ArrowDropDown from "@material-ui/icons/ArrowDropDown";
+import ArrowRight from "@material-ui/icons/ArrowRight";
+import Button from "@material-ui/core/Button";
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import Popover, { PopoverOrigin } from "@material-ui/core/Popover";
+import { usePopover, PopoverElements } from "../../utils/usePopover";
+
+const ANCHOR_ORIGIN: PopoverOrigin = {
+  vertical: "top",
+  horizontal: "right",
+};
+const TRANSFORM_ORIGIN: PopoverOrigin = {
+  vertical: "top",
+  horizontal: "left",
+};
+
+function ListPopover(props: {
+  currentSelection: string;
+  popover: PopoverElements;
+  items: Record<string, string[]> | string[];
+  onClick: (event: React.MouseEvent<HTMLElement>, value: string) => void;
+}) {
+  const hasChildren = !Array.isArray(props.items);
+  const listItems: string[] = hasChildren
+    ? Object.keys(props.items)
+    : (props.items as string[]);
+
+  const renderListItem = (listItem: string) => {
+    if (
+      hasChildren &&
+      (props.items as Record<string, string[]>)[listItem].length === 0
+    ) {
+      return (
+        <ListItem button disabled>
+          {listItem} [unavailable]
+        </ListItem>
+      );
+    } else {
+      return (
+        <ListItem
+          button
+          selected={listItem === props.currentSelection}
+          onClick={(event) => {
+            props.onClick(event, listItem);
+          }}
+        >
+          <ListItemText primary={listItem} />
+          {hasChildren && <ArrowRight />}
+        </ListItem>
+      );
+    }
+  };
+
+  return (
+    <Popover
+      open={props.popover.isOpen}
+      anchorEl={props.popover.anchor}
+      onClose={props.popover.close}
+      anchorOrigin={ANCHOR_ORIGIN}
+      transformOrigin={TRANSFORM_ORIGIN}
+    >
+      <List>
+        {listItems.map((listItem: string) => renderListItem(listItem))}
+      </List>
+    </Popover>
+  );
+}
+
+function TwoLevelDropDown(props: {
+  value: string;
+  options: Record<string, string[]>;
+  onOptionUpdate: (option: string) => void;
+}) {
+  const firstMenu = usePopover();
+  const secondMenu = usePopover();
+
+  const [firstMenuSelection, setFirstMenuSelection] = useState(
+    Object.keys(props.options)[0]
+  );
+
+  const oneLevelMenu = Object.keys(props.options).length === 1;
+
+  return (
+    <>
+      <Button variant="text" onClick={firstMenu.open}>
+        Filter by:<u>{props.value}</u>
+        <ArrowDropDown />
+      </Button>
+
+      <ListPopover
+        popover={firstMenu}
+        currentSelection={firstMenuSelection}
+        items={oneLevelMenu ? Object.values(props.options)[0] : props.options}
+        onClick={(event: React.MouseEvent<HTMLElement>, value: string) => {
+          if (oneLevelMenu) {
+            props.onOptionUpdate(value);
+            firstMenu.close();
```

Comment:
Does this case need to reset the firstMenuSelection state? Or no because it will never be used in a one-level menu?

Reply:
Well I was using it to highlight the currently selected item, but on second thought I think it's better if we don't highlight the current selections. So now it's fine to not reset it since it's only used to render the second menu, which you can only open by selecting a new first menu value.